### PR TITLE
The beutiful blobfish

### PR DESCRIPTION
**Pull request by J35P312**
Blobfish is a wrapper around deseq2. Blobfish accepts a bunch of salmon quant.sf files and  produces all vs all comparisons for each setting/sample.

Visit this page to learn more about BlobFish:

https://www.smithsonianmag.com/science-nature/behold-the-blobfish-180956967/

### CODE REVIEW

File: lib/MIP/Program/Variantcalling/BlobFish.pm [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+## Constants
+Readonly my $SPACE => q{ };
+
+sub blobfish_all_vs_all {
+
+## Function : Perl wrapper for BlobFish --allvsall, version 0.1.0.
+## Returns  : @commands
+## Arguments: $FILEHANDLE             => Filehandle to write to
+##          : $infile_paths_ref       => list of input salmon quant.sf files
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/BlobFish.pm]: list -> List

File: lib/MIP/Program/Variantcalling/BlobFish.pm [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+##          : $in_samples_ref         => list of input sample ids, these should match the infile path list
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/BlobFish.pm]: list -> List

File: lib/MIP/Program/Variantcalling/BlobFish.pm [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+##          : $outdir_path            => Output Directory path
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/BlobFish.pm]: Directory -> directory

File: lib/MIP/Program/Variantcalling/BlobFish.pm [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+##          : $stderrfile_path        => Stderrfile path
+##          : $stderrfile_path_append => Append stderr info to file path
+##          : $stdoutfile_path        => Stdoutfile path
+##          : $tx_to_gene_path            => the path to the transcripts to gene conversion file
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/BlobFish.pm]: - Align fat comma
- the path -> Path
- to gene conversion file -> gene conversion file
- tx? What does it stand for? transcripts?

File: lib/MIP/Program/Variantcalling/BlobFish.pm (hunk 2 above)
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/BlobFish.pm]: in_samples - insamples - throughout module

File: lib/MIP/Program/Variantcalling/BlobFish.pm [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            stderrfile_path        => $stderrfile_path,
+            stderrfile_path_append => $stderrfile_path_append,
+            stdoutfile_path        => $stdoutfile_path,
+        }
+      );
+
+    unix_write_to_file(
+        {
+            FILEHANDLE   => $FILEHANDLE,
+            commands_ref => \@commands,
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/BlobFish.pm]: Order AB

File: t/blobfish_all_vs_all.t [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+          . catfile(qw{ dir A2 quant.sf })
+          . $SPACE
+          . catfile(qw{ dir B1 quant.sf })
+          . $SPACE
+          . catfile(qw{ dir B2 quant.sf }),
+    },
+    in_samples_ref => {
+        inputs_ref      => [ q{A}, q{A}, q{B}, q{B} ],
+        expected_output => q{--sample}
+          . $SPACE . q{B}
```
Comment (henrikstranneheim) [inline t/blobfish_all_vs_all.t]: Does this really pass? Should it not be A, A, B, B?

Supply it in the specifics and see if it passes?

File: lib/MIP/Program/Variantcalling/BlobFish.pm (hunk 4 above)
Comment (J35P312) [reply lib/MIP/Program/Variantcalling/BlobFish.pm]: Yep, tx is for transcripts! it is named like that in the R package that BlowFish wraps around, so I decided to keep it here as well =P

File: t/blobfish_all_vs_all.t (hunk 6 above)
Comment (J35P312) [reply t/blobfish_all_vs_all.t]: Woopsiedaisy, I forgot to put it in specifics. Now it looks passes the test!

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**